tests/handlers/test_appservice.py

```diff
@@ -321,3 +325,241 @@ def _mkservice_alias(self, is_interested_in_alias):
         service.token = "mock_service_token"
         service.url = "mock_service_url"
         return service
+
+
+class ApplicationServiceEphemeralEventsTestCase(unittest.HomeserverTestCase):
+    servlets = [
+        synapse.rest.admin.register_servlets_for_client_rest_resource,
+        login.register_servlets,
+        room.register_servlets,
+        sendtodevice.register_servlets,
+        receipts.register_servlets,
+    ]
+
+    def prepare(self, reactor, clock, hs):
+        # Mock the application service scheduler so that we can track any
+        # outgoing transactions
```

Comment:
given we're mocking this bit out, it might be better to call the class `AppserviceHandlerEphemeralEventsTestCase`. Also worth adding a docstring explaining what exactly is, or is not, in scope.

I hate having to read all the way through a testclass to understand what it's trying to test.